app/models/manageiq/providers/hawkular/middleware_manager/alert_manager.rb

```diff
@@ -0,0 +1,85 @@
+module ManageIQ::Providers
+  class Hawkular::MiddlewareManager::AlertManager
+    require 'hawkular/hawkular_client'
+
+    def initialize(ems)
+      @alerts_client = ems.connect.alerts
+    end
+
+    def process_alert(operation, miq_alert)
+      group_trigger = convert_group_trigger(miq_alert)
+      group_conditions = convert_group_conditions(miq_alert)
+      case operation
+      when :new
+        @alerts_client.create_group_trigger(group_trigger)
+        @alerts_client.set_group_conditions(group_trigger.id,
+                                            :FIRING,
+                                            group_conditions)
+      when :update
+        @alerts_client.update_group_trigger(group_trigger)
+        @alerts_client.set_group_conditions(group_trigger.id,
+                                            :FIRING,
+                                            group_conditions)
+      when :delete
+        @alerts_client.delete_group_trigger(group_trigger.id)
+      end
+    end
+
+    def convert_group_trigger(miq_alert)
+      ::Hawkular::Alerts::Trigger.new('id'          => "MiQ-#{miq_alert[:id]}",
+                                      'name'        => miq_alert[:description],
+                                      'description' => miq_alert[:description],
+                                      'enabled'     => miq_alert[:enabled],
+                                      'type'        => :GROUP,
+                                      'eventType'   => :EVENT)
+    end
+
+    def convert_group_conditions(miq_alert)
+      eval_method = miq_alert[:conditions][:eval_method]
+      options = miq_alert[:conditions][:options]
+      case eval_method
+      when "mw_accumulated_gc_duration"       then generate_mw_gc_conditions(eval_method, options)
+      when "mw_heap_used", "mw_non_heap_used" then generate_mw_jvm_conditions(eval_method, options)
+      end
+    end
+
+    def generate_mw_gc_conditions(eval_method, options)
```

Comment:
iiic this method creates a single condition `c`, but its name is in plural (condition`s`). if that's correct, please rename it.